components/BookCatalog/AddBookModal/AddNewCopyOfBookForm.tsx

```diff
@@ -57,24 +59,45 @@ type addNewCopyOfBookFormProps = {
   author?: string;
 };
 
-const sendBook = async (e) => {
-  e.preventDefault();
-  const url = getEnvUrl();
-  const { title, author, genre, region } = e.target;
-  const book = {
-    title: title.value,
-    author: author.value,
-    checkedOut: false,
-    lastCheckedOut: null,
-    location: 'PEPO Library',
-    genres: [genre.value],
-    regions: [region.value]
+const AddNewCopyOfBookForm = (props: addNewCopyOfBookFormProps) => {
+  const { data, setData } = useContext(BooksContext);
+
+  const sendBook = async (e) => {
+    e.preventDefault();
+    const url = getEnvUrl();
+    const { title, author, genre, region } = e.target;
+    const book = {
+      title: title.value,
+      author: author.value,
+      checkedOut: false,
+      lastCheckedOut: null,
+      location: 'PEPO Library',
+      genres: [genre.value],
+      regions: [region.value]
+    };
+
+    const response = await axios.post(`${url}/api/books`, book);
+    console.log(response);
+
+    const localUpdate = {
+      id: response.data.id,
+      Title: response.data.title,
+      Author: response.data.author,
+      Genres: response.data.genres || [],
+      Regions: response.data.regions || [],
+      'Checked Out': response.data.checkedOut || false,
+      'Checked Out By': 'N/A',
+      'Last Checked Out': formatDate(response.data.lastCheckedOut),
+      Location: response.data.location.name,
+      'Barcode ID': response.data.barcodeId
+    };
```

Comment:
I'm a bit tired so dumb question: Is this meant to update the local table after the response is received? If so, why not just refresh the page? There probably is a reason but I figured I should ask.

Reply:
I thought it would be a smoother user experience if the local data got updated, versus making another call to the database, or refreshing the page. But I see how this is repeating a lot of code. In this case, which option should we go with?